packages/flutter/lib/src/widgets/widget_inspector.dart

```diff
@@ -646,58 +803,203 @@ class WidgetInspectorService {
     return false;
   }
 
-  Map<String, Object> _serializeToJson(DiagnosticsNode node, String groupName) {
-    return _nodeToJson(node, groupName);
+  /// Wrapper around `json.encode` that uses a ring of cached values to prevent
+  /// the Dart garbage collector from collecting objects between when
+  /// the value is returned over the Observatory protocol and when the
+  /// separate observatory protocol command has to be used to retrieve its full
+  /// contents.
```

Comment:
This sounds like a workaround for a VM bug. Is there a bug#?

Reply:
There is now. Filed https://github.com/dart-lang/sdk/issues/32919
and mentioned it in the code.
This is generally by design in the VM service but it would be nice to fix as this also impacts robustness in simple watch windows displaying the values of dart expressions that could be GCed.